solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/impl/ValueFeature.java

```diff
@@ -74,26 +74,31 @@ public FeatureWeight createWeight(IndexSearcher searcher, boolean needsScores, S
 
   public class ValueFeatureWeight extends FeatureWeight {
 
-    final protected float featureValue;
+    final protected Float featureValue;
 
     public ValueFeatureWeight(IndexSearcher searcher, String name,
         NamedParams params, Normalizer norm, int id, SolrQueryRequest request, Query originalQuery, Map<String,String> efi) {
       super(ValueFeature.this, searcher, name, params, norm, id, request, originalQuery, efi);
       if (configValueStr != null) {
         final String expandedValue = macroExpander.expand(configValueStr);
-        if (expandedValue == null) {
-          throw new FeatureException(this.getClass().getCanonicalName()+" requires efi parameter that was not passed in request.");
+        if (expandedValue != null) {
```

Comment:
The code looks good.  The only issue I see is that this will now always silently fail if an efi is not passed in.  In the cases you brought up you want this behavior, but I can see some cases where we do expect an efi to always be passed in.  
You should allow some way to control making a feature's efi required vs optional (possibly through the individual feature's config).